File: java/api/src/main/java/org/dressdiscover/api/models/user/UserId.java
package org.dressdiscover.api.models.user;

import org.thryft.waf.api.models.IntegerModelId;

public final class UserId extends IntegerModelId {
    public static UserId parse(final String value) throws InvalidUserIdException {
        try {
            return new UserId(Integer.parseInt(value));
        } catch (final IllegalArgumentException e) {
            throw InvalidUserIdException.create("not an integer");
        }
    }

    public UserId(final int value) {
        super(value);
    }
}

File: thrift/src/dressdiscover/api/models/user/user_id.py
import os.path, sys
sys.path.append(os.path.join(os.path.dirname(__file__), '..'))
from _id import _Id


class UserId(_Id):
    def java_qname(self, **kwds):
        return 'org.dressdiscover.api.models.user.UserId'

    def java_read_protocol_throws_checked(self):
        return ['org.dressdiscover.api.models.user.InvalidUserIdException']
